src/main/java/org/cryptomator/linux/keychain/KDEWalletKeychainAccess.java

```diff
@@ -80,6 +63,28 @@ public ConnectedWallet(DBusConnection connection) {
 			this.wallet = new KDEWallet(connection);
 		}
 
+		static Optional<ConnectedWallet> connect() {
+			try {
+				return Optional.of(new ConnectedWallet(getConnection()));
+			} catch (RuntimeException | DBusException e) {
+				LOG.warn("Connecting to D-Bus failed.", e);
+				return Optional.empty();
+			}
+		}
+
+		private static DBusConnection getConnection() throws RuntimeException, DBusException {
+			try {
+				return DBusConnection.getConnection(DBusConnection.DBusBusType.SESSION);
+			} catch (RuntimeException e) {
+				if ("Cannot Resolve Session Bus Address".equals(e.getMessage())) {
```

Comment:
This is very prone to unnoticed changes in upstream exception handling. If the message changes, the code stops working.

Can we instead get add an explicit (checked) exception to DBusConnection?

Reply:
This is not possible, see https://github.com/hypfvieh/dbus-java/issues/128.

This issue is already handled in dbus-java (without a new release), but the changed also needs to find its way into kdewallet.

Reply:
Ok. Would you say we can wait with this PR until we get an upstream version without the RuntimeException?